clients/src/main/java/org/apache/kafka/common/record/SkipKeyValueDefaultRecord.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.record;
+
+import org.apache.kafka.common.header.Header;
+
+import java.nio.ByteBuffer;
+
+public class SkipKeyValueDefaultRecord extends DefaultRecord {
```

Comment:
I think the only benefit of extending `DefaultRecord` is that it lets us use a single loop inside `validateMessagesAndAssignOffsetsCompressed`. The downside is that this is not really a valid implementation of `Record`. I think @ijuma suggested the alternative of creating a `PartialRecord` which only exposed the fields we have available here. I think this could work without the need to duplicate the logic in that loop. Though even if we had to duplicate some of the logic (which we already did in `readFromSkipKeyValue`), it might still be preferable in order to keep interface integrity.

Reply:
if `PartialRecord` does not implement `Record` interface how could we avoid duplicating the logic in the loop of `LogValidator`?

Reply:
@hachikuji I looked into how to duplicate code if `SkipKeyValueDefaultRecord` does not extend DefaultRecord, and the amount of duplication is actually pretty large since not only I need to duplicate the loop, but also called private functions like `validateTimestamp`, `validateKey`, `validateRecord`.

I've also tried to have an `interface PartialRecord` and let `Record extends PartialRecord`, but it turns out to be also quite clumsy as I need to refactor lots of the iterator code if I want to de-dup the main loop. 

And I checked the unit test and I think it should be well covering the case if we ever forget about this and added new check that requires these fields, multiple of the existing `LogValidatorTest` cases will crash and fail. So I'd still suggest we go with this approach.